src/v/redpanda/admin_server.cc

```diff
@@ -3452,6 +3454,89 @@ admin_server::cloud_storage_usage_handler(
     }
 }
 
+namespace {
+
+ss::json::json_return_type
+fill_status(cluster::consensus_ptr raft, model::ntp ntp) {
+    if (!raft->is_leader()) {
+        throw ss::httpd::server_error_exception(
+          fmt_with_ctx(fmt::format, "Can not find leader for ntp {}", ntp));
+    }
+
```

Comment:
This should be a redirect to the leader, not exception

Reply:
But it was already redirected. And pass check at the beggining of the handler. So it was leader. But smth goes wrong

Reply:
this is a natural situation, since the leadership information in cache is eventually consistent. Definitely should not be logged with an error and should return 503